### PR TITLE
RFC: remove JuMP from tests and bump versions

### LINKED ISSUE #86: tests fail on 1.2 (dependency issue?)
It looks like this package brings in an older version of JuMP during testing, which fails to load on 1.2 due to the invalid `:import` expression thing that was fixed a while ago in JuMP. Can the test dependencies just be updated or is there a deeper problem?

### PR DESCRIPTION
**Pull request by odow**
I went for the nuclear option, and dropped Julia 0.6 and JuMP 0.18 testing (although MPB) is still there. Testing MPB requires that we keep JuMP 0.18 which isn't great. Moreover, the MOI solver goes through almost all the same code-paths, so the MINLPTests should comprehensively catch most cases.

Closes #86

### CODE REVIEW

File: test/nl_convert.jl [hunk 1]
```diff
@@ -55,21 +53,17 @@ end
...
-    @test_throws Exception solve(m)
-
-    # clean up temp file
-    for file in readdir(AmplNLWriter.solverdata_dir)
-        if file != ".gitignore"
-            rm(joinpath(AmplNLWriter.solverdata_dir, file))
-        end
-    end
-end
+# @testset "[nl_convert] check user defined functions error" begin
```
Comment (mlubin) [inline test/nl_convert.jl]: Why is this commented out (and not removed)?

File: Project.toml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+MathOptInterface = "b8f27783-ece8-5eb3-8dc8-9495eed66fee"
+MathProgBase = "fdba3010-5040-5b88-9595-932c9decdf73"
+SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
+
+[compat]
+Ipopt = "≥ 0.1.15"
+JuMP = "~0.19.0"
+MathOptInterface = "~0.8.2"
+MathProgBase = "~0.5.0, ~0.6, ~0.7"
+julia = "≥ 1.0.0"
```
Comment (blegat) [inline Project.toml]: I tend to prefer `julia = "1"` so that it's not compatible with Julia v2.0

File: Project.toml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+[extras]
+Ipopt = "b6b21f68-93f8-5de0-b562-5493be1d77c9"
+JuMP = "4076af6c-e467-56ae-b986-b466b2749572"
```
Comment (blegat) [inline Project.toml]: Isn't JuMP removed from test ?

**Comment by odow** [comment]
It's actually less breaking that I originally thought it was going to be. I did wonder about removing/deprecating all of the `XXXNLSolver` functions, and deprecate the exported `getxxx` functions.

**Pull request merged**